js_modules/dagster-ui/packages/ui-core/src/app/Util.tsx

```diff
@@ -93,10 +93,14 @@ export const formatElapsedTimeWithoutMsec = (msec: number) => {
 
 export const formatElapsedTimeWithMsec = (msec: number) => {
   const {hours, minutes, seconds, milliseconds} = timeByParts(msec);
+
+  const hasHourOrMinutes = hours || minutes;
+  const hasHoursOrMinutesOrSeconds = hours || minutes || seconds;
+
   const negative = msec < 0;
-  return `${negative ? '-' : ''}${hours}:${twoDigit(minutes)}:${twoDigit(
-    seconds,
-  )}${formatMsecMantissa(milliseconds)}`;
+  return `${negative ? '-' : ''}${hours ? hours + ':' : ''}${
+    hasHourOrMinutes ? twoDigit(minutes) + ':' : ''
+  }${hasHoursOrMinutesOrSeconds ? twoDigit(seconds) : '0'}${formatMsecMantissa(milliseconds)}`;
```

Comment:
`01.069` doesn't look correct to me. I don't think you want two-digit rendering for the first part of the string, whichever part it may be. How about:

```
const sign = negative ? '-' : '';
const hourStr = hours > 0 ? `${hours}:` : '';
const minuteStr = hours > 0 ? `${twoDigit(minutes)}:` : minutes > 0 ? `${minutes}:` : '';
const secStr = hours > 0 || minutes > 0 ? `${twoDigit(seconds)}:` : `${seconds}:`;
const mantissa = formatMsecMantissa(milliseconds);

return `${sign}${hourStr}${minuteStr}${secStr}${manitssa}`;
```